packages/yew/src/dom_bundle/btag/attributes.rs

```diff
@@ -135,25 +145,58 @@ impl Attributes {
                 Some(old) => old != new,
                 None => true,
             } {
-                el.set_attribute(k, new).unwrap();
+                // todo: set property
+                el.set_attribute(k, new.0).unwrap();
             }
         }
 
         // Remove missing
-        for k in old.keys() {
+        for (k, (_, apply_as)) in old.iter() {
             if !new.contains_key(k) {
-                Self::remove_attribute(el, k);
+                Self::remove(el, k, *apply_as);
             }
         }
     }
 
-    fn set_attribute(el: &Element, key: &str, value: &str) {
-        el.set_attribute(key, value).expect("invalid attribute key")
+    fn set(el: &Element, key: &str, value: &str, apply_as: ApplyAttributeAs) {
+        match apply_as {
+            ApplyAttributeAs::Attribute => {
+                el.set_attribute(key, value).expect("invalid attribute key")
+            }
+            ApplyAttributeAs::Property => {
+                match key {
+                    // need to be attributes because, otherwise query selectors fail
+                    "class" => el.set_attribute(key, value).expect("invalid attribute key"),
```

Comment:
Instead of special casing this at runtime, would it be better to do this handling in the macro?

To explain, HTML attributes are _reflected_ by properties on the `HtmlElement`. The complication is that this mapping changes some names.
- `:class` is reflected by [`.className`](https://developer.mozilla.org/en-US/docs/Web/API/Element/className). Imo, `html! { <div class="foo" /> }` could be rewritten - handling this mismatch - into `html! { <div .className="foo /> }`.
- some other attributes that change their name while reflected as a property

The approach would be that the syntax `name` would rewrite some several well-known attribute names into their proper property names, and assign those at runtime instead. `.name` would bypass this rewriting and just assign to the property as named. At least, afaik, the list of attributes that get reflected by the samely named property is small. I can't find a good list though to link here.

Another special case (why is this so messy?... I hate browsers) is the `value` attribute. This is reflected by the `defaultValue` property because, well, it only assigns the default value of an `<input>`. The developer expection (and the current implementation, I believe) is that it dynamically assigns the `value` _property_ instead (and bypasses the `value` attribute completely). Similarly with [`:selected`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option#attr-selected) is reflected as [`.defaultSelected`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOptionElement#htmloptionelement.defaultselected) but I think dev expectation of dynamically binding to [`.selected`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLOptionElement#htmloptionelement.selected) is preferred. See also #2530

Reply:
I removed the special casing entirely as the properties are no longer set by default. They are explicitly opt-in so it's user's responsibility to set the correct one.